datafusion/physical-expr/src/aggregate/first_last.rs

```diff
@@ -611,8 +634,20 @@ impl LastValueAccumulator {
                 }
             })
             .collect::<Vec<_>>();
-        let indices = lexsort_to_indices(&sort_columns, Some(1))?;
-        Ok((!indices.is_empty()).then_some(indices.value(0) as _))
+
+        if self.ignore_nulls {
+            let indices = lexsort_to_indices(&sort_columns, None)?;
+            // If ignoring nulls, find the last non-null value.
+            for index in indices.iter().flatten() {
+                if !value.is_null(index as usize) {
+                    return Ok(Some(index as usize));
+                }
+            }
+            Ok(None)
+        } else {
+            let indices = lexsort_to_indices(&sort_columns, Some(1))?;
+            Ok((!indices.is_empty()).then_some(indices.value(0) as _))
```

Comment:
Why we get the first index here, but not the last one 😕

Reply:
For `last_value` sort expressions are reversed. By this trick, we can get the first expression in the sort to get last one.